### ISSUE TITLE
[Bug] Error: ENOTDIR: not a directory, rmdir './node_modules' 

### ISSUE THREAD

**Issue by joker-777** [yarnpkg/berry#4326]
### Self-service

- [x] I'd be willing to implement a fix

### Describe the bug

Because of our deployment process, we have multiple folders which should share the same `node_modules` folder. Unfortunately, yarn 3.2.0 doesn't support that the `node_modules` is a soft link.

```
yarn
➤ YN0000: ┌ Resolution step
➤ YN0000: └ Completed
➤ YN0000: ┌ Fetch step
➤ YN0013: │ js-tokens@npm:4.0.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ loose-envify@npm:1.4.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ react-dom@npm:18.0.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ react@npm:18.0.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ scheduler@npm:0.21.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0000: └ Completed
➤ YN0000: ┌ Link step
➤ YN0001: │ Error: ENOTDIR: not a directory, rmdir '/home/user/test_app/app/node_modules'
➤ YN0000: └ Completed
➤ YN0000: Failed with errors in 0s 113ms
```

### To reproduce

```
git clone https://github.com/joker-777/test_app
cd test_app/app
yarn
```

### Environment

```shell
System:
    OS: Linux 5.11 Ubuntu 20.04.3 LTS (Focal Fossa)
    CPU: (4) x64 Intel(R) Xeon(R) CPU @ 2.80GHz
  Binaries:
    Node: 14.18.2 - /tmp/xfs-5e35bc1d/node
    Yarn: 3.2.0 - /tmp/xfs-5e35bc1d/yarn
    npm: 6.14.15 - /usr/bin/npm
  npmPackages:
    jest: ^27.5.1 => 27.5.1
```

**Comment by larixer** [comment]
Duplicate of #4172, fixed via: #4206 in master, to verify, run:
`yarn set version from sources`
and then
`yarn`

**Comment by joker-777** [comment]
@larixer Thanks, this definitely fixed the error but there is still a problem. It still creates a new `node_modules` folder instead of using the existing sumlink to our shared `node_modules` folder.

**Comment by larixer** [comment]
@joker-777 Hmm, okay, I see that something is not right, I will take a look and reply later.

**Comment by joker-777** [comment]
@larixer I'm updating right now my example app and there it somehow works. I'll keep investigating.

**Comment by joker-777** [comment]
@larixer Sorry for may false alarm. Everything works. Do know when you guys will publish a new version?

**Comment by arcanis** [comment]
> @larixer Sorry for may false alarm. Everything works. Do know when you guys will publish a new version?

We're currently working on the next major, and cherry-picks are reserved to critical bugfixes. Will take some time.

**Comment by merceyz** [comment]
The fix has been backported and released in v3.2.1
